csharp/src/Ice/Internal/UdpTransceiver.cs

```diff
@@ -726,12 +692,11 @@ internal UdpTransceiver(
         ProtocolInstance instance,
         string host,
         int port,
-        string mcastInterface,
-        bool connect)
+        string mcastInterface)
     {
         _endpoint = endpoint;
         _instance = instance;
-        _state = connect ? StateNeedConnect : StateNotConnected;
+        _state = StateNotConnected;
```

Comment:
A server transceiver (incoming = true) starts in the StateNotConnected state and remains in that state.
A client transceiver (incoming = false) starts in the StateNeedConnect state and goes up to StateConnected. It never reaches StateNotConnected.